src/main/java/liquibase/nosql/database/AbstractNoSqlDatabase.java

```
@@ -0,0 +1,329 @@
+    }
+
+    @Override
+    public String escapeColumnNameList(final String columnNames) {
+        return null;
+    }
+
+    @Override
+    public boolean supportsTablespaces() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsCatalogs() {
+        return true;
+    }
+
+    @Override
+    public CatalogAndSchema.CatalogAndSchemaCase getSchemaAndCatalogCase() {
+        return CatalogAndSchema.CatalogAndSchemaCase.ORIGINAL_CASE;
+    }
+
+    @Override
+    public boolean supportsSchemas() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsCatalogInObjectName(final Class<? extends DatabaseObject> type) {
+        return false;
+    }
+
+    @Override
+    public String generatePrimaryKeyName(final String tableName) {
+        return null;
+    }
+
+    @Override
+    public String escapeSequenceName(final String catalogName, final String schemaName, final String sequenceName) {
+        return null;
+    }
+
+    @Override
+    public String escapeViewName(final String catalogName, final String schemaName, final String viewName) {
+        return null;
+    }
+
+    @Override
+    public String escapeStringForDatabase(final String string) {
+        return null;
+    }
+
+    @Override
+    public abstract void dropDatabaseObjects(final CatalogAndSchema schemaToDrop) throws LiquibaseException;
+
+    @Override
+    public boolean supportsRestrictForeignKeys() {
+        return false;
+    }
+
+    @Override
+    public String escapeConstraintName(final String constraintName) {
+        return null;
+    }
+
+    @Override
+    public List<DatabaseFunction> getDateFunctions() {
+        // irrelevant (will never be called as this is not SQL being processed)
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean supportsForeignKeyDisable() {
+        return false;
+    }
+
+    @Override
+    public boolean disableForeignKeyChecks() {
+        return false;
```

should throw an exception to be aligned with enable below?